pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -1256,6 +1256,20 @@ func (cr *PerconaServerMongoDB) CompareMongoDBVersion(version string) (int, erro
 	return mongoVer.Compare(compare), nil
 }
 
+func (cr *PerconaServerMongoDB) ComparePBMAgentVersion(version string) (int, error) {
+	backupVer, err := v.NewVersion(cr.Status.BackupVersion)
+	if err != nil {
+		return 0, errors.Wrap(err, "parse status.backupVersion")
+	}
+
+	compare, err := v.NewVersion(version)
```

Comment:
I'm thinking that this function is confusing about what is happening with the version comparison. Propose we rename `backupVer` to `existingBackupVer` and this to `expectedBackupVer` instead of `compare` which doesn't make sense. Then the latter comparison will be `existingBackupVer.Compare(expectedBackupVer)`.